components/browser/menu/src/main/java/mozilla/components/browser/menu/item/AbstractParentBrowserMenuItem.kt

```diff
@@ -16,11 +16,14 @@ import mozilla.components.browser.menu.BrowserMenuItem
  * @param endOfMenuAlwaysVisible when is set to true makes sure the bottom of the menu is always visible
  * otherwise, the top of the menu is always visible.
  * @param isCollapsingMenuLimit Whether this menu item can serve as the limit of a collapsing menu.
+ * @param isSticky whether this item menu should not be scrolled offscreen (downwards or upwards
+ * depending on the menu position).
  */
 abstract class AbstractParentBrowserMenuItem(
     private val subMenu: BrowserMenu,
     private val endOfMenuAlwaysVisible: Boolean,
-    override val isCollapsingMenuLimit: Boolean = false
+    override val isCollapsingMenuLimit: Boolean = false,
+    override val isSticky: Boolean = false
```

Comment:
I am thinking maybe we want default `= true` here, since we don't usually want things to scroll offscreen

Reply:
Same for: 
- BrowserMenuCategory.kt
- BrowserMenuCheckbox.kt
- BrowserMenuCheckbox.kt
- BrowserMenuCompoundButton.kt
- BrowserMenuHighlightableSwitch.kt
- BrowserMenuHighlightableItem.kt

Reply:
.. in all of the `BrowserMenuItem`s 😅 I realized there's a bunch more lol

Reply:
If all menu items fit the screen the list is not scrollable by default.
If not all menu items fit the screen they will have to be scrolled off the screen.
Having a sticky item is the exception - a sticky footer will not be scrolled downwards offscreen but all the items that do not fit can be scrolled off to the bottom.
I think the video in the PR description - especially the menu shown in landscape showcases best where the StickyLinearLayoutManager is useful and it's basically the fix for https://github.com/mozilla-mobile/fenix/issues/18591#issuecomment-804841491.